src/symbols/Cluster.tsx

```diff
@@ -133,12 +153,41 @@ export const Cluster: FC<ClusterProps> = ({
     [theme.cluster?.fill]
   );
 
-  useCursor(active && onClick !== undefined, 'pointer');
+  const addDraggingId = useStore(state => state.addDraggingId);
+  const removeDraggingId = useStore(state => state.removeDraggingId);
+  const setClusterPosition = useStore(state => state.setClusterPosition);
+
+  const bind = useDrag({
+    draggable: draggable && !hoveredNodeId,
+    position: {
+      x: position.x,
+      y: position.y,
+      z: -1
+    } as any,
+    set: (pos: Vector3) => setClusterPosition(label, pos as any),
+    onDragStart: () => {
+      addDraggingId(label);
+      setActive(true);
+    },
+    onDragEnd: () => {
+      removeDraggingId(label);
+      setActive(false);
+      onDragged?.({ nodes, label });
+    }
+  });
+
+  useCursor(active && !isDragging && onClick !== undefined, 'pointer');
+  useCursor(
+    active && draggable && !isDraggingCurrent && onClick === undefined,
+    'grab'
+  );
+  useCursor(isDraggingCurrent, 'grabbing');
 
   const { pointerOver, pointerOut } = useHoverIntent({
     disabled,
     onPointerOver: (event: ThreeEvent<PointerEvent>) => {
       setActive(true);
+      cameraControls.controls.truckSpeed = 0;
```

Comment:
We should make a method on cameraControls that will disable this and then restore it to previous value ( rather than assuming 2.0 ).